org.adempiere.base/src/org/compiere/model/MSMTP.java

```diff
@@ -0,0 +1,88 @@
+/***********************************************************************
+ * This file is part of iDempiere ERP Open Source                      *
+ * http://www.idempiere.org                                            *
+ *                                                                     *
+ * Copyright (C) Contributors                                          *
+ *                                                                     *
+ * This program is free software; you can redistribute it and/or       *
+ * modify it under the terms of the GNU General Public License         *
+ * as published by the Free Software Foundation; either version 2      *
+ * of the License, or (at your option) any later version.              *
+ *                                                                     *
+ * This program is distributed in the hope that it will be useful,     *
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of      *
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the        *
+ * GNU General Public License for more details.                        *
+ *                                                                     *
+ * You should have received a copy of the GNU General Public License   *
+ * along with this program; if not, write to the Free Software         *
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,          *
+ * MA 02110-1301, USA.                                                 *
+ *                                                                     *
+ * Contributors:                                                       *
+ * - Nicolas Micoud (TGI)		                                       *
+ **********************************************************************/
+package org.compiere.model;
+
+import java.sql.ResultSet;
+import java.util.Properties;
+
+import org.compiere.util.DB;
+import org.compiere.util.Env;
+
+/**
+ * SMTP servers
+ */
+public class MSMTP extends X_AD_SMTP {
+	/**
+	 * 
+	 */
+	private static final long serialVersionUID = -3645639750095013941L;
+
+	/**
+	 * SMTP constructor
+	 * @param ctx			context
+	 * @param X_AD_SMTP_ID	ID
+	 * @param trxName		transaction
+	 */
+	public MSMTP(Properties ctx, int X_AD_SMTP_ID, String trxName) {
+		super(ctx, X_AD_SMTP_ID, trxName);
+	} // MSMTP
+
+	/**
+	 * SMTP constructor
+	 *
+	 * @param ctx     context
+	 * @param rs      ResultSet
+	 * @param trxName transaction
+	 */
+	public MSMTP(Properties ctx, ResultSet rs, String trxName) {
+		super(ctx, rs, trxName);
+	} // MSMTP
+	
+	/** Search a SMTP server for the sender (1st try with the full address, 2nd try with domain only) */
+	public static int getSmtpID(int clientID, String from, String trxName) {
```

Comment:
- use log instead of System.out

- as you mentioned in the ticket this must be cached

Now, thinking about this approach, if I understand you right you can combine emails and also different domains in the same record - probably you want that separated by something (commas)

But I think that's wrong, if you register carlos@idempiere.com - then that record will be used also for nicolas@idempiere.com because of the domain discovery.

Maybe a better way - and more straight to use with cache - is that the field "Used by emails and/or domains" must be renamed to "Used by email or domain".  And register single record for complete email or for a full domain, that way I think the query must be searched with equal (=) operator instead of LIKE

If several users are intended to send emails with different SMTP then you can register several records.

The method must check first for the EMail (cached) and then check for the domain part (cached too).

Reply:
Hi,
I'm trying to work on this, but i got lots of errors in Eclipse when working on branch 4866 (master is ok).
 (main on is "Failure to find org.eclipse.tycho:tycho-source-plugin:jar:2.4.0-SNAPSHOT"...) ; so I'll try to fix that but if not possible, I think I'll close this PR and create a new one based on latest sources